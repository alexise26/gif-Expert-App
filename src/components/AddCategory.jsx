import { useState } from "react"
import { PropTypes } from "prop-types";

export const AddCategory = ({onNewCategory}) => {
    
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) =>{
        setInputValue(event.target.value);
    }

    const onSubmit = (event) =>{
        event.preventDefault();
        if (inputValue.trim().length <= 1)return;
        // setCategories( cat => [... cat, inputValue]); Esta linea se usa cuando se manda el setcategories del UseState
        onNewCategory(inputValue); //Aqui recibimos la funcion y le decimos que agregar enviando el param
        setInputValue('');
    }
 
    return (
        <form onSubmit={onSubmit} aria-label="form">

            <input 
                type="text"
                placeholder="Find gifs..." 
                value={ inputValue }
                onChange={ onInputChange }
                />

        </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}


