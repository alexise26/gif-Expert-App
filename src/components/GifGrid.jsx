import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { PropTypes } from "prop-types";

export function GifGrid({category}) {
    
    const {images, isLoading} = useFetchGifs( category );// Llamamos nuestro custom hook
  
    return (
    <>
        <h3>{category}</h3>
        {
            isLoading ? (<h2 className="">Cargando...</h2>):null
        }
        <div className="card-grid">
            {   //Mapeamos el resultado [ara sacar el ID
                images.map( (image) => ( 
                    <GifItem 
                        key={image.id} 
                        {...image} //Esto es para enviar todas las props de image en lugar de mandar una por unaßß
                    />
                ))
            }
        </div>

    </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}