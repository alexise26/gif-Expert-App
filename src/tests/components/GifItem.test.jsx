import { GifItem } from "../../components/GifItem";
import { fireEvent, getByText, render, screen } from "@testing-library/react";


describe('Tests para el componente GifItem', () => {

    const images = [
        {id:1,title: 'image1', url:'https://image.com/'},
        {id:2,title: 'image2', url:'https://image.com/'},
        {id:3,title: 'image3', url:'https://image.com/'},
    ];
    const image = images[0];
    
    test('debe coincidir con el snapshot', () => {
        const {container} = render ( <GifItem {...image} /> );
        expect(container).toMatchSnapshot();

    });

    test('debe mostrar la imagen con el URL y ALT correcto', () => {
        render ( <GifItem {...image} /> );
        const {src, alt} = screen.getByRole('img');
        
        expect(alt).toBe(image.title);
        expect(src).toBe(image.url);

    });

    test('debe mostrar el titulo en el componente', () => {
        render ( <GifItem {...image} />);
        
        expect(screen.getByText(image.title)).toBeTruthy();
    });
    
    
});
