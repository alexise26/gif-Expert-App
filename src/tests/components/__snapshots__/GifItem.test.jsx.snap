// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests para el componente GifItem debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="image1"
      src="https://image.com/"
    />
    <p>
      image1
    </p>
  </div>
</div>
`;
